Sources/CMakeLists.txt

```diff
@@ -6,6 +6,32 @@
 # See http://swift.org/LICENSE.txt for license information
 # See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 
+# cmake generation for Swift adds an order only dependency, which matches how C-family languages
+# works. In that case, however, ninja (and presumably other generators) will rebuild on header
+# changes. That's not the case for Swift, and thus if A -> B, A is not being rebuilt when the
+# ABI/API of B changes.
+#
+# For now workaround this by touching a file whenever B is rebuilt and then compiling that file as
+# part of A. Ideally this file would be generated by each of the targets, but that dependency didn't
+# seem to be being tracked.
+function(add_forced_dependencies TARGET)
+  cmake_parse_arguments(ARGS "" "" "PUBLIC" ${ARGN})
+
+  foreach(DEPENDENCY ${ARGS_PUBLIC})
+    target_link_libraries(${TARGET} PUBLIC
```

Comment:
You shouldn't need to add these separately. `target_link_libraries(${TARGET} PUBLIC ${ARGS_PUBLIC})` should work.

Reply:
Ah yeah, sure. For some background this used to be `add_forced_dependency` singular and I changed it to take multiple right before pushing :P. Hence all the force pushes.